mediator/med.F90

```diff
@@ -2128,14 +2120,39 @@ subroutine DataInitialize(gcomp, rc)
             end if
             deallocate(fldnames)
          end if
+      end if
 
-         ! Create field bundles for mediator atm/ocn flux computation
-         call med_phases_aofluxes_init_fldbuns(gcomp, rc=rc)
-         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+      !---------------------------------------
+      ! Initialize field bundles needed for atm/ocn flux computation:
+      ! is_local%wrap%FBMed_aoflux_a and is_local%wrap%FBMed_aoflux_o
+      !---------------------------------------
 
+      ! NOTE: this section must be done BEFORE the second call to esmFldsExchange
+      ! Create field bundles for mediator ocean albedo computation
+
+      fieldCount = med_fldList_GetNumFlds(fldListMed_aoflux)
+      if ( fieldCount > 0 ) then
+         if ( is_local%wrap%med_coupling_active(compocn,compatm) .or. &
+              is_local%wrap%med_coupling_active(compatm,compocn)) then
+            is_local%wrap%do_med_aoflux = .true.
```

Comment:
Is this logic correct? `is_local%wrap%do_med_aoflux` is set as `true` in here but in line 2148 it sets to `false`. There is nothing between them related with setting `do_med_aoflux`.

Reply:
@uturuncoglu - you are right. It turns out that is_local%wrap%do_med_aoflux is not used any more and so I have now removed it. I have just pushed my changes back with this removed. The rest of the logic should be correct.